test-container/src/main/java/io/strimzi/StrimziKafkaContainer.java

```diff
@@ -133,13 +142,23 @@ protected void containerIsStarting(InspectContainerResponse containerInfo, boole
             kafkaListenerSecurityProtocol.append(",");
         });
 
+        StringBuilder kafkaConfiguration = new StringBuilder();
+
+        // default listener config
+        kafkaConfiguration.append(" --override listeners=" + kafkaListeners + "PLAINTEXT://0.0.0.0:" + KAFKA_PORT);
+        kafkaConfiguration.append(" --override advertised.listeners=" + advertisedListeners);
+        kafkaConfiguration.append(" --override zookeeper.connect=localhost:" + ZOOKEEPER_PORT);
+        kafkaConfiguration.append(" --override listener.security.protocol.map=" + kafkaListenerSecurityProtocol + "PLAINTEXT:PLAINTEXT");
+        kafkaConfiguration.append(" --override inter.broker.listener.name=BROKER1");
+
+        // additional kafka config
+        kafkaConfigurationMap.forEach((configName, configValue) -> {
+            kafkaConfiguration.append(" --override " + configName + "=" + configValue);
```

Comment:
Validate that none of the `configName` is one of the hard coded configs.

Also, this could be a `.map(...).forEach(kafkaConfiguration::append)`, I think.

Reply:
I don't want to forbid users from these options. For instance, if the user wants to override`listeners` or `zookeeper.connect` config is ok or?

Reply:
Fair enough, but does `--override` support later options overriding previous ones?

Reply:
Yes It does )